packages/core/admin/admin/src/content-manager/pages/ListView/components/BulkActionButtons/index.js

```diff
@@ -69,16 +77,75 @@ const confirmDialogsPropTypes = {
  * ConfirmDialogPublishAll
  * -----------------------------------------------------------------------------------------------*/
 
-const ConfirmDialogPublishAll = ({ isOpen, onToggleDialog, isConfirmButtonLoading, onConfirm }) => {
+export const ConfirmDialogPublishAll = ({
+  isOpen,
+  onToggleDialog,
+  isConfirmButtonLoading,
+  onConfirm,
+}) => {
   const { formatMessage } = useIntl();
+  const { get } = useFetchClient();
+  const { selectedEntries } = useTableContext();
+  const toggleNotification = useNotification();
+  const { formatAPIError } = useAPIErrorHandler(getTrad);
+  const {
+    contentType: { uid: slug },
+  } = useSelector(listViewDomain());
+
+  const {
+    data: countDraftRelations,
+    isLoading,
+    isError,
+    error,
+  } = useQuery(
+    ['content-manager', 'draft-relations', slug, selectedEntries],
+    async () => {
+      /**
+       * TODO: refactor the API call when the param serializer will be ready
+       */
+      const {
+        data: { data },
+      } = await get(
+        `/content-manager/collection-types/${slug}/actions/countManyEntriesDraftRelations?${stringify(
+          { ids: selectedEntries },
+          { encode: false }
+        )}`
+      );
+
+      return data;
+    },
+    {
+      enabled: isOpen,
+    }
+  );
+
+  if (isError) {
+    toggleNotification({ type: 'warning', message: formatAPIError(error) });
+
+    return null;
+  }
```

Comment:
this concerns me, can you walk me through how you think this is going to work? 🤔

Reply:
Yes, of course. If I don't add this condition, the ConfirmDialogPublishAll will still appear even if there is an error coming from the API. By adding this condition, I prevent the ConfirmDialogPublishAll from being rendered in the presence of an error.

Reply:
But wouldn't this toggle a notification every time the component renders?

Reply:
Here we're performing a side effect and if the component re-renders through react (for whatever reason that maybe) then even if it's fired, it'll fire again – i'm wondering if the side-effect should be performed in a useEffect to avoid this multiple firing event 🤔  wdyt?

Reply:
Do you mean something like this?

Reply:
I changed how I handle the error in this way, let me know if it is ok for you, thanks
```
  const {
    data: countDraftRelations,
    isLoading,
    isError,
  } = useQuery(
    ['content-manager', 'draft-relations', slug, selectedEntries],
    async () => {
      /**
       * TODO: refactor the API call when the param serializer will be ready
       */
      const {
        data: { data },
      } = await get(
        `/content-manager/collection-types/${slug}/actions/countManyEntriesDraftRelations?${stringify(
          { ids: selectedEntries },
          { encode: false }
        )}`
      );

      return data;
    },
    {
      enabled: isOpen,
      onError(error) {
        toggleNotification({ type: 'warning', message: formatAPIError(error) });
      },
    }
  );

  if (isError) {
    return null;
  }
```